2A

In [ ]:
from google.colab import files

uploaded = files.upload()

for fn in uploaded.keys():
  print('User uploaded file "{name}" with length {length} bytes'.format(
      name=fn, length=len(uploaded[fn])))

KeyboardInterrupt: ignored

In [ ]:
!unzip "iris_multimodal_dataset.zip"

In [ ]:
import cv2
import os
from google.colab.patches import cv2_imshow

In [ ]:
import pandas as pd
import numpy as np
import pickle
import matplotlib.pyplot as plt
import seaborn as sns
import matplotlib.style as style
import ast
from sklearn.preprocessing import MultiLabelBinarizer

from gensim.models import KeyedVectors

In [ ]:
df = pd.read_csv("/content/iris_multimodal_dataset/iris_multimodal.csv")

In [ ]:
df

sepal_length  sepal_width  ...         species           image
0             5.1          3.5  ...     Iris-setosa  00ea5d35aa.jpg
1             4.9          3.0  ...     Iris-setosa  09b4632cbf.jpg
2             4.7          3.2  ...     Iris-setosa  0ef9fffaf5.jpg
3             4.6          3.1  ...     Iris-setosa  15cdc80d57.jpg
4             5.0          3.6  ...     Iris-setosa  1a37baefcf.jpg
..            ...          ...  ...             ...             ...
145           6.7          3.0  ...  Iris-virginica  d724d9d0cb.jpg
146           6.3          2.5  ...  Iris-virginica  dd555866c6.jpg
147           6.5          3.0  ...  Iris-virginica  dfa937cb81.jpg
148           6.2          3.4  ...  Iris-virginica  f2afd59439.jpg
149           5.9          3.0  ...  Iris-virginica  f45f73cc01.jpg

[150 rows x 6 columns]

In [ ]:
df[['sepal_length',	'sepal_width',	'petal_length',	'petal_width']].astype('float')

sepal_length  sepal_width  petal_length  petal_width
0             5.1          3.5           1.4          0.2
1             4.9          3.0           1.4          0.2
2             4.7          3.2           1.3          0.2
3             4.6          3.1           1.5          0.2
4             5.0          3.6           1.4          0.2
..            ...          ...           ...          ...
145           6.7          3.0           5.2          2.3
146           6.3          2.5           5.0          1.9
147           6.5          3.0           5.2          2.0
148           6.2          3.4           5.4          2.3
149           5.9          3.0           5.1          1.8

[150 rows x 4 columns]

In [ ]:
df['sepal_length'].dtypes

dtype('float64')

In [ ]:
df['sepal_width'].dtypes

dtype('float64')

In [ ]:
df['petal_length'].dtypes

dtype('float64')

In [ ]:
df['petal_width'].dtypes

dtype('float64')

In [ ]:
# from sklearn import preprocessing
# le = preprocessing.LabelEncoder()
# le.fit(df["species"])
# print(le.classes_)
# le.transform(df["species"])

In [ ]:
from keras.utils import np_utils
from sklearn import preprocessing

In [ ]:
encoder =  preprocessing.LabelEncoder()
encoder.fit(df["species"])
encoded_Y = encoder.transform(df["species"])
# convert integers to dummy variables (i.e. one hot encoded)


In [ ]:
df["species"]=encoded_Y

In [ ]:
df["species"]

0      0
1      0
2      0
3      0
4      0
      ..
145    2
146    2
147    2
148    2
149    2
Name: species, Length: 150, dtype: int64

Taking each one of the species


In [ ]:
filter = df['species'] == 0
dfsetosa = df[filter]
filter = df['species'] == 1
dfversi = df[filter]
filter = df['species'] == 2
dfvirgin = df[filter]


In [ ]:
dfsetosa

sepal_length  sepal_width  ...  species           image
0            5.1          3.5  ...        0  00ea5d35aa.jpg
1            4.9          3.0  ...        0  09b4632cbf.jpg
2            4.7          3.2  ...        0  0ef9fffaf5.jpg
3            4.6          3.1  ...        0  15cdc80d57.jpg
4            5.0          3.6  ...        0  1a37baefcf.jpg
5            5.4          3.9  ...        0  32d84c9c2f.jpg
6            4.6          3.4  ...        0  35af907c72.jpg
7            5.0          3.4  ...        0  3885ce6d63.jpg
8            4.4          2.9  ...        0  3f7da4fc31.jpg
9            4.9          3.1  ...        0  57e4915544.jpg
10           5.4          3.7  ...        0  59268cb7a8.jpg
11           4.8          3.4  ...        0  5b3cb8f13b.jpg
12           4.8          3.0  ...        0  5c4ecfb6bb.jpg
13           4.3          3.0  ...        0  618f3ee446.jpg
14           5.8          4.0  ...        0  633e4ad1e5.jpg
15           5.7          4.4  ...        0  667f0f3fd4.jpg
16           5.4          3.9  ...        0  68f71330c0.jpg
17           5.1          3.5  ...        0  6e99afe9a3.jpg
18           5.7          3.8  ...        0  74c3a969df.jpg
19           5.1          3.8  ...        0  7595364873.jpg
20           5.4          3.4  ...        0  7652de79df.jpg
21           5.1          3.7  ...        0  76f001f8ee.jpg
22           4.6          3.6  ...        0  7b9efd8dcf.jpg
23           5.1          3.3  ...        0  7d296418da.jpg
24           4.8          3.4  ...        0  81ad98f884.jpg
25           5.0          3.0  ...        0  830fea9655.jpg
26           5.0          3.4  ...        0  8a8ba48b39.jpg
27           5.2          3.5  ...        0  92f1f8baea.jpg
28           5.2          3.4  ...        0  92f8b21728.jpg
29           4.7          3.2  ...        0  931ce43060.jpg
30           4.8          3.1  ...        0  94677c3c29.jpg
31           5.4          3.4  ...        0  9a5a0d8e74.jpg
32           5.2          4.1  ...        0  9a79832266.jpg
33           5.5          4.2  ...        0  9aad38bce8.jpg
34           4.9          3.1  ...        0  a40341151c.jpg
35           5.0          3.2  ...        0  a88919feb1.jpg
36           5.5          3.5  ...        0  bb5c53aa2a.jpg
37           4.9          3.1  ...        0  bf5a5b27f4.jpg
38           4.4          3.0  ...        0  ca3080a237.jpg
39           5.1          3.4  ...        0  ca8b11c4eb.jpg
40           5.0          3.5  ...        0  cbebe60d0d.jpg
41           4.5          2.3  ...        0  cdbf830aaa.jpg
42           4.4          3.2  ...        0  cfe7d30ccb.jpg
43           5.0          3.5  ...        0  da41891f1c.jpg
44           5.1          3.8  ...        0  e311bbfaeb.jpg
45           4.8          3.0  ...        0  e4ff929dd3.jpg
46           5.1          3.8  ...        0  e886e8c7a3.jpg
47           4.6          3.2  ...        0  ee3d85d3e3.jpg
48           5.3          3.7  ...        0  ef042f3ca0.jpg
49           5.0          3.3  ...        0  fcf886ed50.jpg

[50 rows x 6 columns]

In [ ]:
dfversi

sepal_length  sepal_width  ...  species           image
50           7.0          3.2  ...        1  0a1b9a8a98.jpg
51           6.4          3.2  ...        1  0a5831fcf8.jpg
52           6.9          3.1  ...        1  0fceb55da9.jpg
53           5.5          2.3  ...        1  131ffec81f.jpg
54           6.5          2.8  ...        1  209dcc6064.jpg
55           5.7          2.8  ...        1  22178f5aca.jpg
56           6.3          3.3  ...        1  3223960fa8.jpg
57           4.9          2.4  ...        1  451be4a90d.jpg
58           6.6          2.9  ...        1  4eab48940f.jpg
59           5.2          2.7  ...        1  5ea23e98aa.jpg
60           5.0          2.0  ...        1  6068cc6cf1.jpg
61           5.9          3.0  ...        1  616cd67a39.jpg
62           6.0          2.2  ...        1  667a8243a8.jpg
63           6.1          2.9  ...        1  6e7982bbf9.jpg
64           5.6          2.9  ...        1  70c23852de.jpg
65           6.7          3.1  ...        1  722fde7c98.jpg
66           5.6          3.0  ...        1  7a43ed360d.jpg
67           5.8          2.7  ...        1  7ab4f2df7d.jpg
68           6.2          2.2  ...        1  7ba1416246.jpg
69           5.6          2.5  ...        1  7dd6d5fed5.jpg
70           5.9          3.2  ...        1  807f09a571.jpg
71           6.1          2.8  ...        1  837979a22c.jpg
72           6.3          2.5  ...        1  8380f9f054.jpg
73           6.1          2.8  ...        1  847b45809f.jpg
74           6.4          2.9  ...        1  8c12d19fa0.jpg
75           6.6          3.0  ...        1  8d4dbc1218.jpg
76           6.8          2.8  ...        1  9d3b685f50.jpg
77           6.7          3.0  ...        1  9ec9d92eb7.jpg
78           6.0          2.9  ...        1  9f79c4e746.jpg
79           5.7          2.6  ...        1  a3e41664bc.jpg
80           5.5          2.4  ...        1  ac9fa3588b.jpg
81           5.5          2.4  ...        1  acca9c0de0.jpg
82           5.8          2.7  ...        1  af5bf183c9.jpg
83           6.0          2.7  ...        1  b1b8d816da.jpg
84           5.4          3.0  ...        1  b3eec67a3a.jpg
85           6.0          3.4  ...        1  b4e7f044e7.jpg
86           6.7          3.1  ...        1  bef4ae7602.jpg
87           6.3          2.3  ...        1  bfce6cf99b.jpg
88           5.6          3.0  ...        1  c24b9266f6.jpg
89           5.5          2.5  ...        1  c3ac5af864.jpg
90           5.5          2.6  ...        1  c3e9a10fdb.jpg
91           6.1          3.0  ...        1  c4c16ac325.jpg
92           5.8          2.6  ...        1  c63ec146ea.jpg
93           5.0          2.3  ...        1  c761b51252.jpg
94           5.6          2.7  ...        1  c8e6435979.jpg
95           5.7          3.0  ...        1  c95520fff0.jpg
96           5.7          2.9  ...        1  c9cb770ffe.jpg
97           6.2          2.9  ...        1  d0e28a0d8b.jpg
98           5.1          2.5  ...        1  d4ed5e2fa9.jpg
99           5.7          2.8  ...        1  dc4a211d41.jpg

[50 rows x 6 columns]

In [ ]:
dfvirgin

sepal_length  sepal_width  ...  species           image
100           6.3          3.3  ...        2  00b7e0fc2a.jpg
101           5.8          2.7  ...        2  044c00f694.jpg
102           7.1          3.0  ...        2  0a5f520707.jpg
103           6.3          2.9  ...        2  1dd4139c8f.jpg
104           6.5          3.0  ...        2  200e376483.jpg
105           7.6          3.0  ...        2  255d42041a.jpg
106           4.9          2.5  ...        2  2fa37a88aa.jpg
107           7.3          2.9  ...        2  30310e9cf9.jpg
108           6.7          2.5  ...        2  3162b10b53.jpg
109           7.2          3.6  ...        2  3e89d249bd.jpg
110           6.5          3.2  ...        2  4391fe1b69.jpg
111           6.4          2.7  ...        2  458632c9e3.jpg
112           6.8          3.0  ...        2  4800f42af0.jpg
113           5.7          2.5  ...        2  4eee33988e.jpg
114           5.8          2.8  ...        2  526c6450cd.jpg
115           6.4          3.2  ...        2  5971b001db.jpg
116           6.5          3.0  ...        2  5a940173e8.jpg
117           7.7          3.8  ...        2  6485a9f041.jpg
118           7.7          2.6  ...        2  64f29561c5.jpg
119           6.0          2.2  ...        2  6524bf3658.jpg
120           6.9          3.2  ...        2  74f680a275.jpg
121           5.6          2.8  ...        2  75e9d01f0e.jpg
122           7.7          2.8  ...        2  766b6a5e3b.jpg
123           6.3          2.7  ...        2  79aec67c42.jpg
124           6.7          3.3  ...        2  890db0c0e1.jpg
125           7.2          3.2  ...        2  8b09942388.jpg
126           6.2          2.8  ...        2  8ce9e8155b.jpg
127           6.1          3.0  ...        2  9305bd55ac.jpg
128           6.4          2.8  ...        2  939aa86920.jpg
129           7.2          3.0  ...        2  99d35f624d.jpg
130           7.4          2.8  ...        2  9ab5f154c9.jpg
131           7.9          3.8  ...        2  9c656dabf3.jpg
132           6.4          2.8  ...        2  a881752ddc.jpg
133           6.3          2.8  ...        2  a96a1b9067.jpg
134           6.1          2.6  ...        2  ab0aba91e9.jpg
135           7.7          3.0  ...        2  b73efb7a84.jpg
136           6.3          3.4  ...        2  c0e83f3c90.jpg
137           6.4          3.1  ...        2  cad93560ce.jpg
138           6.0          3.0  ...        2  cc7c7d5f0d.jpg
139           6.9          3.1  ...        2  cd0c9545bb.jpg
140           6.7          3.1  ...        2  cedea707c9.jpg
141           6.9          3.1  ...        2  cf63b9c6a7.jpg
142           5.8          2.7  ...        2  d3ed088ddd.jpg
143           6.8          3.2  ...        2  d4a2c31ffc.jpg
144           6.7          3.3  ...        2  d5d7aa9cb1.jpg
145           6.7          3.0  ...        2  d724d9d0cb.jpg
146           6.3          2.5  ...        2  dd555866c6.jpg
147           6.5          3.0  ...        2  dfa937cb81.jpg
148           6.2          3.4  ...        2  f2afd59439.jpg
149           5.9          3.0  ...        2  f45f73cc01.jpg

[50 rows x 6 columns]

In [ ]:
df

sepal_length  sepal_width  ...  species           image
0             5.1          3.5  ...        0  00ea5d35aa.jpg
1             4.9          3.0  ...        0  09b4632cbf.jpg
2             4.7          3.2  ...        0  0ef9fffaf5.jpg
3             4.6          3.1  ...        0  15cdc80d57.jpg
4             5.0          3.6  ...        0  1a37baefcf.jpg
..            ...          ...  ...      ...             ...
145           6.7          3.0  ...        2  d724d9d0cb.jpg
146           6.3          2.5  ...        2  dd555866c6.jpg
147           6.5          3.0  ...        2  dfa937cb81.jpg
148           6.2          3.4  ...        2  f2afd59439.jpg
149           5.9          3.0  ...        2  f45f73cc01.jpg

[150 rows x 6 columns]

Splitting dataset into 30% testing and 70% trainning

In [ ]:
from sklearn.model_selection import train_test_split
df_train_seto, df_test_seto = train_test_split(dfsetosa, test_size=0.3)

In [ ]:
df_train_versi, df_test_versi = train_test_split(dfversi, test_size=0.3)

In [ ]:
df_train_virgin, df_test_virgin = train_test_split(dfvirgin, test_size=0.3)

Train Combine (From Iris Sentosa, Iris Versicolor, and Iris Virginica)

In [ ]:
x_train = df_train_seto[['sepal_length','sepal_width','petal_length','petal_width','image']]

In [ ]:
x_train=x_train.append(df_train_versi[['sepal_length','sepal_width','petal_length','petal_width','image']])

In [ ]:
x_train=x_train.append(df_train_virgin[['sepal_length','sepal_width','petal_length','petal_width','image']])

In [ ]:
x_train.shape

(105, 5)

In [ ]:
x_train_img=x_train['image']

In [ ]:
x_train_img

25     830fea9655.jpg
49     fcf886ed50.jpg
45     e4ff929dd3.jpg
16     68f71330c0.jpg
4      1a37baefcf.jpg
            ...      
124    890db0c0e1.jpg
144    d5d7aa9cb1.jpg
119    6524bf3658.jpg
110    4391fe1b69.jpg
142    d3ed088ddd.jpg
Name: image, Length: 105, dtype: object

In [ ]:
x_train.pop("image")

25     830fea9655.jpg
49     fcf886ed50.jpg
45     e4ff929dd3.jpg
16     68f71330c0.jpg
4      1a37baefcf.jpg
            ...      
124    890db0c0e1.jpg
144    d5d7aa9cb1.jpg
119    6524bf3658.jpg
110    4391fe1b69.jpg
142    d3ed088ddd.jpg
Name: image, Length: 105, dtype: object

In [ ]:
x_train

sepal_length  sepal_width  petal_length  petal_width
25            5.0          3.0           1.6          0.2
49            5.0          3.3           1.4          0.2
45            4.8          3.0           1.4          0.3
16            5.4          3.9           1.3          0.4
4             5.0          3.6           1.4          0.2
..            ...          ...           ...          ...
124           6.7          3.3           5.7          2.1
144           6.7          3.3           5.7          2.5
119           6.0          2.2           5.0          1.5
110           6.5          3.2           5.1          2.0
142           5.8          2.7           5.1          1.9

[105 rows x 4 columns]

In [ ]:
y_train = df_train_seto['species']

In [ ]:
y_train = y_train.append(df_train_versi['species'])

In [ ]:
y_train = y_train.append(df_train_virgin['species'])

In [ ]:
y_train= np_utils.to_categorical(y_train)

In [ ]:
y_train.shape

(105, 3)

Test Combine

In [ ]:
x_test = df_test_seto[['sepal_length','sepal_width','petal_length','petal_width','image']]
x_test=x_test.append(df_test_versi[['sepal_length','sepal_width','petal_length','petal_width','image']])
x_test=x_test.append(df_test_virgin[['sepal_length','sepal_width','petal_length','petal_width','image']])

In [ ]:
x_test.shape

(45, 5)

In [ ]:
x_test_img=x_test['image']

In [ ]:
x_test_img

30     94677c3c29.jpg
24     81ad98f884.jpg
15     667f0f3fd4.jpg
38     ca3080a237.jpg
29     931ce43060.jpg
43     da41891f1c.jpg
11     5b3cb8f13b.jpg
44     e311bbfaeb.jpg
21     76f001f8ee.jpg
0      00ea5d35aa.jpg
18     74c3a969df.jpg
32     9a79832266.jpg
23     7d296418da.jpg
10     59268cb7a8.jpg
37     bf5a5b27f4.jpg
59     5ea23e98aa.jpg
87     bfce6cf99b.jpg
76     9d3b685f50.jpg
54     209dcc6064.jpg
55     22178f5aca.jpg
95     c95520fff0.jpg
70     807f09a571.jpg
94     c8e6435979.jpg
73     847b45809f.jpg
71     837979a22c.jpg
80     ac9fa3588b.jpg
63     6e7982bbf9.jpg
51     0a5831fcf8.jpg
68     7ba1416246.jpg
96     c9cb770ffe.jpg
122    766b6a5e3b.jpg
127    9305bd55ac.jpg
129    99d35f624d.jpg
114    526c6450cd.jpg
139    cd0c9545bb.jpg
112    4800f42af0.jpg
130    9ab5f154c9.jpg
135    b73efb7a84.jpg
136    c0e83f3c90.jpg
121    75e9d01f0e.jpg
108    3162b10b53.jpg
146    dd555866c6.jpg
103    1dd4139c8f.jpg
143    d4a2c31ffc.jpg
133    a96a1b9067.jpg
Name: imag

In [ ]:
x_test.pop('image')

30     94677c3c29.jpg
24     81ad98f884.jpg
15     667f0f3fd4.jpg
38     ca3080a237.jpg
29     931ce43060.jpg
43     da41891f1c.jpg
11     5b3cb8f13b.jpg
44     e311bbfaeb.jpg
21     76f001f8ee.jpg
0      00ea5d35aa.jpg
18     74c3a969df.jpg
32     9a79832266.jpg
23     7d296418da.jpg
10     59268cb7a8.jpg
37     bf5a5b27f4.jpg
59     5ea23e98aa.jpg
87     bfce6cf99b.jpg
76     9d3b685f50.jpg
54     209dcc6064.jpg
55     22178f5aca.jpg
95     c95520fff0.jpg
70     807f09a571.jpg
94     c8e6435979.jpg
73     847b45809f.jpg
71     837979a22c.jpg
80     ac9fa3588b.jpg
63     6e7982bbf9.jpg
51     0a5831fcf8.jpg
68     7ba1416246.jpg
96     c9cb770ffe.jpg
122    766b6a5e3b.jpg
127    9305bd55ac.jpg
129    99d35f624d.jpg
114    526c6450cd.jpg
139    cd0c9545bb.jpg
112    4800f42af0.jpg
130    9ab5f154c9.jpg
135    b73efb7a84.jpg
136    c0e83f3c90.jpg
121    75e9d01f0e.jpg
108    3162b10b53.jpg
146    dd555866c6.jpg
103    1dd4139c8f.jpg
143    d4a2c31ffc.jpg
133    a96a1b9067.jpg
Name: imag

In [ ]:
x_test

sepal_length  sepal_width  petal_length  petal_width
30            4.8          3.1           1.6          0.2
24            4.8          3.4           1.9          0.2
15            5.7          4.4           1.5          0.4
38            4.4          3.0           1.3          0.2
29            4.7          3.2           1.6          0.2
43            5.0          3.5           1.6          0.6
11            4.8          3.4           1.6          0.2
44            5.1          3.8           1.9          0.4
21            5.1          3.7           1.5          0.4
0             5.1          3.5           1.4          0.2
18            5.7          3.8           1.7          0.3
32            5.2          4.1           1.5          0.1
23            5.1          3.3           1.7          0.5
10            5.4          3.7           1.5          0.2
37            4.9          3.1           1.5          0.1
59            5.2          2.7           3.9          1.4
87            6.3          2.3           4.4          1.3
76            6.8          2.8           4.8          1.4
54            6.5          2.8           4.6          1.5
55            5.7          2.8           4.5          1.3
95            5.7          3.0           4.2          1.2
70            5.9          3.2           4.8          1.8
94            5.6          2.7           4.2          1.3
73            6.1          2.8           4.7          1.2
71            6.1          2.8           4.0          1.3
80            5.5          2.4           3.8          1.1
63            6.1          2.9           4.7          1.4
51            6.4          3.2           4.5          1.5
68            6.2          2.2           4.5          1.5
96            5.7          2.9           4.2          1.3
122           7.7          2.8           6.7          2.0
127           6.1          3.0           4.9          1.8
129           7.2          3.0           5.8          1.6
114           5.8          2.8           5.1          2.4
139           6.9          3.1           5.4          2.1
112           6.8          3.0           5.5          2.1
130           7.4          2.8           6.1          1.9
135           7.7          3.0           6.1          2.3
136           6.3          3.4           5.6          2.4
121           5.6          2.8           4.9          2.0
108           6.7          2.5           5.8          1.8
146           6.3          2.5           5.0          1.9
103           6.3          2.9           5.6          1.8
143           6.8          3.2           5.9          2.3
133           6.3          2.8           5.1          1.5

In [ ]:
y_test = df_test_seto['species']
y_test = y_test.append(df_test_versi['species'])
y_test = y_test.append(df_test_virgin['species'])

In [ ]:
y_test= np_utils.to_categorical(y_test)

In [ ]:
y_test.shape

(45, 3)

Reading Image and apply it ino x_train_img and x_test_img

In [ ]:
def image_process(src,path):
  images=[]
  for i in src:
    img = cv2.imread(os.path.join(path,i))
    img = cv2.resize(img, (75, 115))
    img = img.astype(np.float32)/255
    images.append(img)
  return images

#src = df["image"]
#path=/content/iris_multimodal_dataset/images

In [ ]:
x_train_img=image_process(x_train_img,"/content/iris_multimodal_dataset/images")

In [ ]:
x_test_img=image_process(x_test_img,"/content/iris_multimodal_dataset/images")

In [ ]:

from keras.preprocessing.text import Tokenizer
from keras.preprocessing.sequence import pad_sequences
from keras.models import Model, load_model
from keras.layers import Embedding, LSTM, Dropout, Dense, Input, Bidirectional, Flatten, Conv2D, MaxPooling2D, concatenate, Conv1D, MaxPooling1D
import keras.backend as K
from keras.optimizers import Adam, RMSprop
from keras.callbacks import ModelCheckpoint, CSVLogger, EarlyStopping

Change Image into Array

In [ ]:
x_train_img=np.array(x_train_img)

In [ ]:
x_test_img=np.array(x_test_img)

Deep Learning model combination of CNN and BPNN

In [ ]:
num_input = Input(shape=(x_train.shape[1],),name='main_input')
num = Dense(32)(num_input)
num_out=Dense(3,activation='sigmoid',name='num_output')(num)


cnn_input = Input(shape=(x_train_img.shape[1], x_train_img.shape[2], x_train_img.shape[3]),name='cnn_input' )
y = Conv2D(32, (3, 3), activation='relu', input_shape=(x_train_img.shape[1], x_train_img.shape[2], x_train_img.shape[3]))(cnn_input)
y = MaxPooling2D(2, 2)(y)
y = Conv2D(64, (3, 3), activation='relu')(y)
y = MaxPooling2D(2, 2)(y)
y = Conv2D(128, (3, 3), activation='relu')(y)
y = MaxPooling2D(2, 2)(y)
y = Conv2D(128, (3, 3), activation='relu')(y)
y = MaxPooling2D(2, 2)(y)
y = Flatten()(y)
y = Dropout(0.3)(y)
cnn_out = Dense(3, activation='relu')(y)

x=concatenate([num_out,cnn_out])
x = Dense(64, activation='relu')(x)
x = Dense(64, activation='relu')(x)
x = Dense(64, activation='relu')(x)

main_output = Dense(3, activation='sigmoid', name='main_output')(x)



In [ ]:
model = Model(inputs=[num_input,cnn_input],outputs=main_output)
model.summary

<bound method Model.summary of <tensorflow.python.keras.engine.functional.Functional object at 0x7ff19f98f908>>

In [ ]:
y_train

array([[1., 0., 0.],
       [1., 0., 0.],
       [1., 0., 0.],
       [1., 0., 0.],
       [1., 0., 0.],
       [1., 0., 0.],
       [1., 0., 0.],
       [1., 0., 0.],
       [1., 0., 0.],
       [1., 0., 0.],
       [1., 0., 0.],
       [1., 0., 0.],
       [1., 0., 0.],
       [1., 0., 0.],
       [1., 0., 0.],
       [1., 0., 0.],
       [1., 0., 0.],
       [1., 0., 0.],
       [1., 0., 0.],
       [1., 0., 0.],
       [1., 0., 0.],
       [1., 0., 0.],
       [1., 0., 0.],
       [1., 0., 0.],
       [1., 0., 0.],
       [1., 0., 0.],
       [1., 0., 0.],
       [1., 0., 0.],
       [1., 0., 0.],
       [1., 0., 0.],
       [1., 0., 0.],
       [1., 0., 0.],
       [1., 0., 0.],
       [1., 0., 0.],
       [1., 0., 0.],
       [0., 1., 0.],
       [0., 1., 0.],
       [0., 1., 0.],
       [0., 1., 0.],
       [0., 1., 0.],
       [0., 1., 0.],
       [0., 1., 0.],
       [0., 1., 0.],
       [0., 1., 0.],
       [0., 1., 0.],
       [0., 1., 0.],
       [0., 1., 0.],
       [0., 1

In [ ]:
model.compile(loss='binary_crossentropy', optimizer='adam', metrics=['accuracy'])

In [ ]:
model.fit([x_train, x_train_img], y_train,epochs=10,validation_data=([x_test,x_test_img],y_test))

Epoch 1/10
4/4 [==============================] - 1s 368ms/step - loss: 0.0811 - accuracy: 0.9714 - val_loss: 0.2499 - val_accuracy: 0.8222
Epoch 2/10
4/4 [==============================] - 1s 349ms/step - loss: 0.1029 - accuracy: 0.9714 - val_loss: 0.1813 - val_accuracy: 0.9111
Epoch 3/10
4/4 [==============================] - 1s 342ms/step - loss: 0.0987 - accuracy: 0.9429 - val_loss: 0.1522 - val_accuracy: 0.9111
Epoch 4/10
4/4 [==============================] - 1s 341ms/step - loss: 0.0633 - accuracy: 0.9714 - val_loss: 0.1507 - val_accuracy: 0.8889
Epoch 5/10
4/4 [==============================] - 1s 340ms/step - loss: 0.0404 - accuracy: 1.0000 - val_loss: 0.1924 - val_accuracy: 0.9111
Epoch 6/10
4/4 [==============================] - 1s 340ms/step - loss: 0.0479 - accuracy: 1.0000 - val_loss: 0.1463 - val_accuracy: 0.8667
Epoch 7/10
4/4 [==============================] - 1s 351ms/step - loss: 0.0382 - accuracy: 1.0000 - val_loss: 0.2322 - val_accuracy: 0.8667
Epoch 8/10
4/4 [====

https://towardsdatascience.com/multimodal-deep-learning-to-predict-movie-genres-e6855f814a8a

https://towardsdatascience.com/multimodal-deep-learning-ce7d1d994f4

2B

Dari hasil trainning diatas dapat dilihat bahwa hasil dari training tersebut memiliki accuracy 100% dengan Validasi accuracy 88%-91% yang dimana menunjukkan bahwa ketepatan dari model ini cukup tinggi.

2C
Keunikan dari model deep learning ini adalah dengan menggunakan gabungan dari CNN dan Connected neural network dapat menghasilkan model dengan accuracy yang cukup tinggi